src/plugins/controls/public/control_types/options_list/options_list_editor.tsx

```diff
@@ -60,7 +60,7 @@ export const OptionsListEditor = ({
         initialInput?.dataViewId ?? getRelevantDataViewId?.() ?? (await getDefaultId());
       let dataView: DataView | undefined;
       if (initialId) {
-        onChange({ dataViewId: initialId });
+        onChange({ dataViewId: initialId, fieldName: initialInput?.fieldName });
```

Comment:
Because we weren't making the `fieldName` part of the initial input on mount for editing existing controls, this meant it was originally `undefined` - so if you changed, for example, just the control label (and did not **specifically** re-click the exact same field), then the selections would be unnecessarily cleared because the original field would not equal `undefined` and therefore ` diffDataFetchProps` would be called. 

By making `fieldName` part of the initial input (when editing an existing control), this is prevented 👍

Reply:
I undid this change, and was unable to reproduce the issue where changing the label and not the fieldName would clear the selections on an _options list control_. I believe this is because the Options List Presave transform does a null check as well as the `deepEqual`.

For an options list, the logic looks like:
 `newInput.fieldName && !deepEqual(newInput.fieldName, embeddable.getInput().fieldName)`,

Whereas in the Range slider and Time Slider the logic looks like
`!deepEqual(newInput.fieldName, embeddable.getInput().fieldName)`

I think it's a bit simpler, and more easily maintainable if this is fixed from within the `presaveTransformFunction`.

Reply:
That's a great point! I implemented the change in `presaveTransformFunction` for each control instead, and it works great :)